### DISCUSSION TITLE
How to create interactive stories

### DISCUSSION THREAD

**Discussion by xeruf** [tmedwards/tweego#38] [Q&A] [answered]
Soo, Twine is about nonlinear stories. I read through the whole documentation, and https://www.motoslave.net/tweego/docs/#twee-notation multiple times, but I do not see how one can use tweego to define any interaction.

An example would be very useful.

**Comment by tmedwards** [comment, marked as answer]
Twine, meaning the entire ecosystem of compilers and story formats, is about interactive fiction.  Linear vs. non-linear isn't really a concern for Twine—that's more a concern for individual projects.

Tweego is a command line compiler for Twee notation files—well, and other file types.  You don't define interaction directly via it anymore than you do with Twine 2 itself.

The most basic form of interaction in Twine is via passage links, which work the same regardless of which compiler you're using.  Beyond that, other forms of interaction generally depend on the story format being used, which also works the same regardless of which compiler you're using.

For example, a basic passage link (in Twee notation):

```
:: Start
This is the start of a long journey.

[[Take the first step!|Oops]]


:: Oops
Oops.  You fell into a hole and died.
```

The passages and how you link them work exactly the same as it does in Twine 2.  You just do it via text files, in Twee notation, rather than Twine 2's GUI.

**Reply by xeruf** [reply]
thanks, I did not know that link format existed in Twee notation, I have never seen it mentioned anywhere?

**Reply by tmedwards** [reply]
It doesn't exist in any compiler, technically, which is why it's not mentioned by the Twee notation specification.

All formatting, markup, macros, etc. come from the story formats.  The basic link markup is simply common to all of them, so authors are least know how to link passages regardless of the format.

Twine 2 only understands the most basic link markup and only enough to generate connection arrows and create new passages.  It doesn't really support or understand them beyond that.

**Reply by tmedwards** [reply]
Ugh.  I really need to stop answering these from e-mail.  It's butchering the formatting.
